### PR TITLE
[RISCV] Ignore ELF flag conflicts for data-only object files.

### PR DESCRIPTION
**Pull request by bsdjhb**
The current set of ELF header flags for RISC-V affect code generation
but not data layout.  Thus, an object file only containing data can be
safely linked into any RISC-V ELF file.  This change ignores the ELF
flags from object files that do not contain any executable sections
when checking for incompatible ELF flags.

### CODE REVIEW

File: lld/ELF/Arch/RISCV.cpp [hunk 1]
```diff
@@ -115,6 +115,24 @@ static uint32_t getEFlags(InputFile *f) {
   return cast<ObjFile<ELF32LE>>(f)->getObj().getHeader()->e_flags;
 }
 
+static bool hasExecutableSection(InputFile *f) {
+  if (config->is64) {
+    auto of = cast<ObjFile<ELF64LE>>(f);
+    auto sections = CHECK(of->getObj().sections(), of);
+    for (const auto section : sections)
+      if (section.sh_flags & SHF_EXECINSTR)
+        return true;
```
Comment (jrtc27) [inline lld/ELF/Arch/RISCV.cpp:124]: Upstream would probably prefer braces round the outer for loop (and repeated below).

Comment (arichardson) [reply lld/ELF/Arch/RISCV.cpp:124]: I guess that depends on the reviewer, I've submitted patches with braces in the past and it was suggested to remove them.

Comment (jrtc27) [reply lld/ELF/Arch/RISCV.cpp:124]: For the `if` I'd leave it without, but for the `for` I'd put them in. Chaining the single expressions feels wrong... and I would be surprised if someone actively made the outer braces be removed (and I'd argue against it myself).

**Comment by arichardson** [comment]
Should we close this PR since we no longer need the feature?

**Pull request closed without merging**